x/stakeibc/keeper/reward_converter_test.go

```diff
@@ -0,0 +1,346 @@
+		},
+	}
+
+	// Create an epoch tracker to dictate the timeout
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx, types.EpochTracker{
+		EpochIdentifier:    epochtypes.STRIDE_EPOCH,
+		NextEpochStartTime: uint64(epochEndTime.UnixNano()),
+		Duration:           uint64(epochLength.Nanoseconds()),
+	})
+
+	// Define the expected transfer message using all the above
+	memoJSON := fmt.Sprintf(`{"forward":{"receiver":"%s","port":"transfer","channel":"%s","timeout":"%s","retries":0}}`,
+		tradeAddress, rewardToTradeChannelId, transfer2TimeoutDuration)
+
+	expectedMsg := transfertypes.MsgTransfer{
+		SourcePort:       transfertypes.PortID,
+		SourceChannel:    hostToRewardChannelId,
+		Token:            transferToken,
+		Sender:           withdrawalAddress,
+		Receiver:         unwindAddress,
+		TimeoutTimestamp: uint64(transfer1TimeoutTimestamp.UnixNano()),
+		Memo:             memoJSON,
+	}
+
+	// Confirm the generated message matches expectations
+	actualMsg, err := s.App.StakeibcKeeper.GetHostToTradeTransferMsg(s.Ctx, transferAmount, route)
+	s.Require().NoError(err, "no error expected when building transfer message")
+	s.Require().Equal(expectedMsg, actualMsg, "transfer message should have matched")
+
+	// Call the main transfer function and confirm the sequence number increments
+	startSequence := s.MustGetNextSequenceNumber(portId, channelId)
+
+	err = s.App.StakeibcKeeper.TransferRewardTokensHostToTrade(s.Ctx, transferAmount, route)
+	s.Require().NoError(err, "no error expected when submitting transfer")
+
+	sequenceAfterTransfer := s.MustGetNextSequenceNumber(portId, channelId)
+	s.Require().Equal(startSequence+1, sequenceAfterTransfer, "sequence number should have incremented")
+
+	// Attempt to call the function again with an transfer amount below the min,
+	// it should not submit an ICA
+	invalidTransferAmount := minSwapAmount.Sub(sdkmath.OneInt())
+	err = s.App.StakeibcKeeper.TransferRewardTokensHostToTrade(s.Ctx, invalidTransferAmount, route)
+	s.Require().NoError(err, "no error expected when submitting transfer with amount below minimum")
+
+	endSequence := s.MustGetNextSequenceNumber(portId, channelId)
+	s.Require().Equal(sequenceAfterTransfer, endSequence, "sequence number should NOT have incremented")
+
+	// Remove the connection ID and confirm the ICA fails
+	invalidRoute := route
+	invalidRoute.HostAccount.ConnectionId = ""
+	err = s.App.StakeibcKeeper.TransferRewardTokensHostToTrade(s.Ctx, transferAmount, invalidRoute)
+	s.Require().ErrorContains(err, "Failed to submit ICA tx")
+
+	// Delete the epoch tracker and call each function, confirming they both fail
+	s.App.StakeibcKeeper.RemoveEpochTracker(s.Ctx, epochtypes.STRIDE_EPOCH)
+
+	_, err = s.App.StakeibcKeeper.GetHostToTradeTransferMsg(s.Ctx, transferAmount, route)
+	s.Require().ErrorContains(err, "epoch not found")
+	err = s.App.StakeibcKeeper.TransferRewardTokensHostToTrade(s.Ctx, transferAmount, route)
+	s.Require().ErrorContains(err, "epoch not found")
+}
```

Comment:
The reason I separate out the tests in the Swap case was because there are a lot of different test cases for different pool configurations that I don't think warrant a full ICA test on. But for this transfer, I'm not sure all that is necessary since it's simpler. Fwiw, this test _does_ test the build function directly as well [[ref](https://github.com/Stride-Labs/stride/pull/1005/files/39c821c69e8b6eed1ce304eeb4b7a1cd1c90acf1#diff-1629187c35b24ec74868f7e895ab32337777eaf56723e246d834f27e31e7d75eR120)]

Reply:
Yep seems good!